lib/storage/metadata/mongoclient/MongoClientInterface.ts

```diff
@@ -381,13 +608,21 @@ class MongoClientInterface {
         });
     }
 
-    putBucketAttributes(bucketName, bucketMD, log, cb) {
+    putBucketAttributes(bucketName: string, bucketMD: BucketInfo, log: werelogs.Logger, cb: NodeCallBack<void>) {
         // FIXME: there should be a version of BucketInfo.serialize()
         // that does not JSON.stringify()
         const bucketInfo = BucketInfo.fromObj(bucketMD);
```

Comment:
is this needed, e.g. is input not already of type `BucketInfo` ?

or maybe we want to keep it anyway, to ensure it works as before called with a plain object with the same fields (since we don't have typescript everywhere) ? then we may be conditional depending on type:

```suggestion
const bucketInfo = bucketMD instanceof BucketInfo ? bucketMD : BucketInfo.fromObj(bucketMD);
```